apps/www/.vitepress/theme/components/CodeSandbox.vue

```diff
@@ -9,13 +9,22 @@ import { type Style } from '@/lib/registry/styles'
 const props = defineProps<{
   name: string
   code: string
+  newYorkCode: string
   style: Style
+  extraFiles?: string[]
 }>()
 
 const sources = ref<Record<string, string>>({})
 
 onMounted(() => {
-  sources.value['App.vue'] = props.code
+  sources.value['App.vue'] = props.style === 'default' ? props.code : props.newYorkCode
+  if (props.extraFiles) {
+    props.extraFiles.forEach((file) => {
+      import(`../../../src/lib/registry/${props.style}/example/${file}.vue?raw`).then((module) => {
+        sources.value[`${file}.vue`] = module.default
+      })
+    })
+  }
```

Comment:
unfortunately this dynamic import wouldn't work, as we are rendering `ComponentPreview` dynamically, and not statically rendered.

Reply:
I think we would need to create a `registry.json` for demos .. so that the code can be imported correctly during production.

Reply:
Seems working here 🤔 

https://github.com/radix-vue/shadcn-vue/assets/17789047/02264df4-19de-4b76-a8e0-dca4fbc4cb7b

Reply:
Hmm.. what's the setup ya? after bundling (pnpm build) it fails to resolve ``../../../src/lib/registry/${props.style}/example/${file}.vue?raw``

Reply:
Tested with `vitepress build` and `vitepress preview` still got the same result